### PR TITLE
'WebUI: adds tracing for webui file downloads #665

### PR DESCRIPTION
**Pull request by sprenner**
'WebUI: adds tracing for webui file downloads #665

### CODE REVIEW

File: lib/rucio/web/ui/static/did.js [hunk 1]
```diff
@@ -181,6 +181,63 @@ load_replicas = function(scope, name) {
...
+    document.body.removeChild(element);
+};
+
+uuid4 = function () {
+    var uuid = '';
+    for (var i = 0; i < 32; i ++) {
+        switch (i) {
+            case 8:
+            case 20:
+                uuid += '-';
```
Comment (tbeerman) [inline lib/rucio/web/ui/static/did.js]: As you are removing the dashes later on anyway, why not just omitting them here in the first place?

File: lib/rucio/web/ui/static/rucio.js [hunk 2]
```diff
@@ -1391,6 +1392,43 @@ RucioClient.prototype.delete_rse_attribute = function(options) {
     });
 };
 
+/* send trace */
+RucioClient.prototype.send_trace = function(options) {
+    check_token();
```
Comment (tbeerman) [inline lib/rucio/web/ui/static/rucio.js]: You don't need the token for the traces, so you don't have to check it here.

File: lib/rucio/web/ui/static/rucio.js [hunk 3]
```diff
@@ -1391,6 +1392,43 @@ RucioClient.prototype.delete_rse_attribute = function(options) {
...
+    var url = this.url + '/traces/';
+    if (options.async == null) { options.async = true; }
+    jQuery.ajax({
+        url: url,
+        async: options.async,
+        crossDomain: true,
+        headers: this.headers,
```
Comment (tbeerman) [inline lib/rucio/web/ui/static/rucio.js]: See above. The token and account are not needed, so you don't have to send them in the header.

**Review by tbeerman — CHANGES_REQUESTED** [review]

File: lib/rucio/web/ui/static/did.js (hunk 1 above)
Comment (sprenner) [reply lib/rucio/web/ui/static/did.js]: To have a function that meets the RFC4122 requirements - but I can omit them here already if you think thats better :)

Comment (tbeerman) [reply lib/rucio/web/ui/static/did.js]: OK I see your point. But here it's not a utility function which will be used anywhere else but rather just there to create the uuid for the traces, which are without dashes.

**Review by tbeerman — APPROVED** [review]

**Pull request merged**